board/safety/safety_nissan.h

```diff
@@ -0,0 +1,188 @@
+
+const uint32_t NISSAN_RT_INTERVAL = 250000;    // 250ms between real time checks
+
+const struct lookup_t NISSAN_LOOKUP_ANGLE_RATE_UP = {
+  {2., 7., 17.},
+  {5., .8, .15}};
+
+const struct lookup_t NISSAN_LOOKUP_ANGLE_RATE_DOWN = {
+  {2., 7., 17.},
+  {5., 3.5, .4}};
+
+const struct lookup_t NISSAN_LOOKUP_MAX_ANGLE = {
+  {3.3, 12, 32},
+  {540., 120., 23.}};
+
+const AddrBus NISSAN_TX_MSGS[] = {{0x169, 0}, {0x20b, 2}};
+
+AddrCheckStruct nissan_rx_checks[] = {
+  {.addr = {0x2}, .bus = 0, .expected_timestep = 100000U},
+  {.addr = {0x29a}, .bus = 0, .expected_timestep = 50000U},
+  {.addr = {0x20b}, .bus = 0, .expected_timestep = 50000U},
+  {.addr = {0x1b6}, .bus = 1, .expected_timestep = 100000U},
+};
+const int NISSAN_RX_CHECK_LEN = sizeof(nissan_rx_checks) / sizeof(nissan_rx_checks[0]);
+
+float nissan_speed = 0;
+int nissan_controls_allowed_last = 0;
+uint32_t nissan_ts_angle_last = 0;
+int nissan_cruise_engaged_last = 0;
+int nissan_desired_angle_last = 0;
+
+struct sample_t angle_meas;            // last 3 steer angles
+
+
+static int nissan_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
+
+  bool valid = addr_safety_check(to_push, nissan_rx_checks, NISSAN_RX_CHECK_LEN,
+                                 NULL, NULL, NULL);
+
+  if (valid) {
+    int bus = GET_BUS(to_push);
+    int addr = GET_ADDR(to_push);
+
+    if (bus == 0) {
+      if (addr == 0x2) {
+        // Current steering angle
+        // Factor -0.1, little endian
+        int angle_meas_new = (GET_BYTES_04(to_push) & 0xFFFF);
+        angle_meas_new = to_signed(angle_meas_new, 16) * -0.1;
+
+        // update array of samples
+        update_sample(&angle_meas, angle_meas_new);
+      }
+
+      if (addr == 0x29a) {
+        // Get current speed
+        // Factor 0.00555
+        nissan_speed = ((GET_BYTE(to_push, 2) << 8) | (GET_BYTE(to_push, 3))) * 0.00555 / 3.6;
+      }
+    }
+
+    if (bus == 1) {
+      if (addr == 0x1b6) {
+        int cruise_engaged = (GET_BYTE(to_push, 4) >> 6) & 1;
+        if (cruise_engaged && !nissan_cruise_engaged_last) {
+          controls_allowed = 1;
+        }
+        if (!cruise_engaged) {
+          controls_allowed = 0;
+        }
+        nissan_cruise_engaged_last = cruise_engaged;
+      }
+    }
+  }
```

Comment:
Missing explicit disable checks on:
- gas rising edge 
- brake pedal pressed when speed > 0

Even though these checks are missing/incomplete in few other safety files, it would be nice to merge this one with those checks already included.